### ISSUE TITLE
Question about connection pooling

### ISSUE THREAD

**Issue by praskutti** [scoriani/reliable-jdbc-app#2]
Hello @scoriani , if you have spare time, can you please take a look at the question I posted about [Connection Pooling in mssql-jdbc driver](https://github.com/microsoft/mssql-jdbc/issues/1513)?

**Comment by scoriani** [comment]
Hi praskutti, you have several ways to check if connection pooling is actually working. You can check it on the server-side, by looking at DMVs like sys.dm_exec_connections and make sure connections are not closed and reopened every time, or by checking at perfcounters (e.g. SELECT * FROM sys.dm_os_performance_counters WHERE counter_name = 'User Connections';).
But you can also enable logging for both JDBC and HikariCP and see the number of connections that are in the pool (active, idle, etc.).

**Comment by praskutti** [comment]
@scoriani , thanks for the guidance. I looked at those queries. Your article is perfect for what I was looking to accomplish - reliable connection from my Spring application using HikariCP + mssql-jdbc driver to connect to SQL 2017 (on premise).
More info: From what I've seen in my application, the cntr_value reduces, but in the trace I still get 1-NonPooled always :( I dont see the sp_reset_connection happening in the profiler trace.

**Comment by scoriani** [comment]
@praskutti, that's correct. There's a note on mssql-jdbc doc page mentioning that sp_reset_connection is not called by the driver itself, hence trace results may not be accurate (at the end of the day, connection pooling is a client thing, and server may or may not be aware). Glad the article was helpful!
